luigi/contrib/bigquery.py

```diff
@@ -507,40 +517,139 @@ def run(self):
         output = self.output()
         assert isinstance(output, BigQueryTarget), 'Output must be a BigQueryTarget, not %s' % (output)
 
+        project_id = output.table.project_id
+
         bq_client = output.client
+        gcs_client = GCSClient()
```

Comment:
This is only needed in the chunked case . Should we hence move it into the if-else block to remove computational overhead in the non-chunked version? (If I am not mistaken instantiating the GCSClient runs the full google-auth procedure?)